### PR TITLE
GUVNOR-2278 : Project Settings: Dependencies should expose dependency scope

### PR DESCRIPTION
**Pull request by Rikkola**
The goal was to add make it possible to edit scope of the dependencies defined in pom.xml, but while I was there...

The reasons for the extra changes:
* Any pom.xml we open should be saved as it was before. Project editor just allows the user to edit a selected set of fields. We do not want to overwrite anything that the user might have hard edited in with a text editor, by passing our guided editor for the pom.xml. 
* Guvnor should not add the kie maven plugin. This leaks drools/jbpm into guvnor and guvnor should be reusable.
* @manstis already did some changes for the NewProjectEditor, but I changed the passing of a GAV to POM. The wizard still edits just the GAV part, but does not care if there are plugins or parent GAV declared for the to be new project.
* If I touched something that was lacking tests I wrote some tests.

https://github.com/droolsjbpm/guvnor/pull/204
https://github.com/droolsjbpm/kie-wb-common/pull/171
https://github.com/droolsjbpm/drools-wb/pull/97

### CODE REVIEW

File: guvnor-project/guvnor-project-api/src/main/java/org/guvnor/common/services/project/model/Dependencies.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+ */
+package org.guvnor.common.services.project.model;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+public class Dependencies
```
Comment (manstis) [inline guvnor-project/guvnor-project-api/src/main/java/org/guvnor/common/services/project/model/Dependencies.java]: Should this be Portable too?

Comment (Rikkola) [inline guvnor-project/guvnor-project-api/src/main/java/org/guvnor/common/services/project/model/Dependencies.java]: For some reason it works as it is, but it should yes. Making it Portable.

File: guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/BuildContentHandler.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+            if ( from.getPlugins() != null ) {
+                to.setPlugins( new PluginUpdater( to.getPlugins() ).update( from.getPlugins() ) );
+            }
+
+            return to;
+        }
+    }
+
+    private Plugin update( org.guvnor.common.services.project.model.Plugin from,
```
Comment (manstis) [inline guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/BuildContentHandler.java]: Is this ever used? (I might be going blind - or was it before ```PluginUpdater``` was used?)

File: guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/PluginUpdater.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+public class PluginUpdater {
+
+    private final List<Plugin> plugins;
+
+    public PluginUpdater( List<Plugin> plugins ) {
+        this.plugins = plugins;
+    }
+
+    public List<Plugin> update( ArrayList<org.guvnor.common.services.project.model.Plugin> from ) {
+
```
Comment (manstis) [inline guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/PluginUpdater.java]: Nice to see you've left some room for future improvements ;)

File: guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/BuildContentHandler.java (hunk 2 above)
Comment (Rikkola) [inline guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/BuildContentHandler.java]: Not used, left over from moving the functionality to another class. Removing it.

File: guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/PluginUpdater.java (hunk 3 above)
Comment (Rikkola) [inline guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/PluginUpdater.java]: Now that I took a look. The method above should be here. Adding test coverage and a fix.

**Comment by Rikkola** [comment]
@manstis changes are done based on comments.

**Comment by manstis** [comment]
+1 to merge.

**Comment by csadilek** [comment]
Thanks.

master: 976e7611c8f809a78b52f1b24f147180bdcf3642

**Pull request closed without merging**